source/routing/raptor_api.cpp

```diff
@@ -1365,6 +1367,65 @@ void filter_late_journeys(RAPTOR::Journeys& journeys, const NightBusFilter::Para
     }
 }
 
+void filter_twisted_journeys(RAPTOR::Journeys& journeys, const bool clockwise) {
+    if (journeys.size() == 1) {
+        return;
+    }
+
+    auto it = journeys.begin();
+    while (it != journeys.end()) {
+        const auto& journey = *it;
+
+        if (!journey.is_pt()) {
+            ++it;
+            continue;
+        }
+        if (journey.sections.size() < 2) {
+            ++it;
+            continue;
+        }
+
+        bool found = false;
+
+        if (clockwise) {
+            auto section_it = journey.sections.rbegin();
+            const auto& last_stop_area_uri = section_it->get_out_st->stop_point->stop_area->uri;
+            for (++section_it; section_it != journey.sections.rend(); ++section_it) {
+                assert(section_it->get_in_st < section_it->get_out_st);
```

Comment:
maybe log an error and exit this function instead of asserting, so that the program don't crash if that happens